sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,40 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainEntryPoint := {
+        if (scalaJSUseMainEntryPoint.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but scalaJSUseMainEntryPoint " +
+                "was set to true. If you do not want to derive a main entry " +
+                "point from discovered main classes, add " +
+                "`scalaJSMainEntryPoint := None`")
+          } { mainClass =>
+            Some(EntryPoint.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
+      },
```

Comment:
What happens if we make this non-optional, but do not pull it in unless `scalaJSUseMainEntryPoint` is true?

Currently, we have the implicit assumption that `scalaJSUseMainEntryPoint == scalaJSMainEntryPoint.isDefined`. It would be nicer if we could get rid of the unneeded degree of freedom. In the worst case, we should just assert that.